lib/msal-node/src/network/HttpClient.ts

```diff
@@ -267,7 +267,7 @@ const networkRequestViaHttps = <T>(
                     // do not destroy the request for the device code flow
                     networkResponse.body["error"] !== Constants.AUTHORIZATION_PENDING) {
                     request.destroy();
-                    reject(new Error(`HTTP status code ${statusCode}`));
+
```

Comment:
removing the early rejection

Reply:
How are testing this fix? You should also delete line 269, which is why I suggested deleting the entire if statement. Additionally, you should also make the same change to the if statement on line 179.

Reply:
Using the Username-Password sample, as described in the original issue. 
Let's chat further. @Robbie-Microsoft

Reply:
> 

The Username-Password sample is throwing the appropriate server error on commenting out L270 alone. 
Still, I have removed the entire if-block as it was mainly a getaround for validateStatus() and the network errors are already handled upstream. 
We can look at adding a similar if-block for network errors based on the returned error object (Or, status code- 5xx /429) in a separate PR in future , if the distinction is desired. I believe the logic for detecting a network error would be different as compared to Axios client. 
@Robbie-Microsoft @jasonnutter